lancero_source.go

```diff
@@ -569,7 +577,15 @@ func (ls *LanceroSource) getNextBlock() chan *dataBlock {
 				panic(fmt.Sprintf("timeout, no data from lancero after %v / %v", panicTime, ls.readPeriod))
 
 			case mfo := <-ls.mixRequests:
-				ls.Mix[mfo.ChannelIndex].errorScale = mfo.MixFraction / float64(ls.nsamp)
+				for i, index := range mfo.ChannelIndices {
+					fraction := mfo.MixFractions[i]
```

Comment:
It deleted my other comment because I just changed this in the code. But I removed the bounds check with silent ignore here. We should just panic on out of bounds and fix the source. Also the bounds are pre-checked in ConfigureMixFaction.

Reply:
I think it's fine to skip checks here, because--as you say--they are already checked in `ConfigureMixFraction`.